File: KHost.App/ClientApp/src/app/models/SongSearchResult.ts
export class SongSearchResult {

    id: string;

    songName: string;

    bandName: string;

    engineName: string;

    lengthInSeconds: number;

    constructor({
        id,
        songName,
        bandName,
        engineName,
        lengthInSeconds
    }: parameters) {
        this.id = id;
        this.songName = songName;
        this.bandName = bandName;
        this.engineName = engineName;
        this.lengthInSeconds = lengthInSeconds;
    }
}

interface parameters {
    id: string,
    songName: string,
    bandName: string,
    engineName: string,
    lengthInSeconds: number
}

File: KHost.App/Models/SongSearchResult.cs
namespace KHost.App.Models
{
    public class SongSearchResult : BaseModel
    {
        public string Id { get; set; }

        public string SongName { get; set; }

        public string BandName { get; set; }

        public string EngineName { get; set; }

        public int? LengthInSeconds { get; set; } 
    }
}
